beacon-chain/execution/service.go

```diff
@@ -819,9 +815,9 @@ func (s *Service) ensureValidPowchainData(ctx context.Context) error {
 			Trie:              s.depositTrie.ToProto(),
 			DepositContainers: s.cfg.depositCache.AllDepositContainers(ctx),
 		}
-		return s.cfg.beaconDB.SaveExecutionChainData(ctx, eth1Data)
+		return nil, s.cfg.beaconDB.SaveExecutionChainData(ctx, eth1Data)
```

Comment:
This becomes an issue as now you have returned a nil object, so in the event of new nodes, they will be initialized with a nil eth1data object and presumably panic.

Reply:
good catch, fixed.